src/components/SelectionList/types.ts

```diff
@@ -360,6 +360,8 @@ type BaseSelectionListProps<TItem extends ListItem> = Partial<ChildrenProps> & {
     /** Custom content to display in the header */
     headerContent?: ReactNode;
 
+    ListHeaderComponent?: React.JSX.Element | null;
```

Comment:
Why is `ListHeaderComponent` CamelCase instead of `listHeaderComponent` as lowerCamelCase?

Additionally, I see `headerContent`, `footerContent`, and `listFooterContent`, so should this be named `listHeaderContent` for consistency?